cypress/integration/search_spec.js

```diff
@@ -0,0 +1,130 @@
+describe('Search', function() {
+  beforeEach(function() {
+    cy.visit('/')
+    cy.get('#filter-bar').as('filterBar')
+    cy.get('#movie-list').as('movieList')
+  })
+
+  it('has "Sort By"', function() {
+    cy.get('@filterBar').contains('Sort By')
+  })
+
+  it('has input', function() {
+    cy.get('@filterBar').find('input')
+  })
+
+  it('has Search button', function() {
+    cy
+      .get('@filterBar')
+      .contains('Search')
+      .should('match', 'button')
+  })
+
+  it('has 10 default results', function() {
+    cy
+      .get('@movieList')
+      .find('.movie-row')
+      .should('have.length', 10)
+  })
+
+  describe('when sorted', function() {
+    context('by Year', function() {
+      beforeEach(function() {
+        cy
+          .get('@filterBar')
+          .contains('Sort By')
+          .click()
+        cy
+          .get('@filterBar')
+          .contains('Year')
+          .click()
+      })
+
+      it('sorts earliest first', function() {
+        cy
+          .get('@movieList')
+          .first('.movie-row')
+          .contains('All Quiet on the Western Front')
```

Comment:
My preference would be for the test to:

1. Aggregate all of the movie years in the default list.
2. `.sort()`.
3. Assert that the new listing is different & sorted.

But, because the markup is plaintext + `<span>` tags, it's very brittle to parse & effort should be put into better markup (possibly using schema.org markers).